agent/api/ecsclient/client.go

```diff
@@ -230,6 +236,34 @@ func (client *APIECSClient) registerContainerInstance(clusterRef string, contain
 	return aws.StringValue(resp.ContainerInstance.ContainerInstanceArn), err
 }
 
+// disableTaskENIEnabledIfNoInit disables the Task ENI capability of the
+// Agent if the Agent was not started with an init system. We need the init system
+// to properly cleanup the dhclient processes when Task ENI capability is
+// enabled
+func (client *APIECSClient) disableTaskENIEnabledIfNoInit() {
+	ok, err := isAgentAlsoTheInit()
+	if err != nil {
+		seelog.Infof("Unable to determine if Agent is started with an init system; Disabling Task ENI capability: %v", err)
+		// TODO: We still need to remove the "task-eni" capability from the
+		// capabilities list. This will be done in the subsequent PR, where
+		// the `TaskEngine.Capabilities()` method will also be refactored
+		// into the `app` package
+		client.config.TaskENIEnabled = false
+		return
+	}
+
+	if !ok {
+		return
+	}
+	seelog.Info("Agent is not started with an init system; Disabling Task ENI capability")
```

Comment:
If it was explicitly enabled, this should probably be a WARN, since we're now disabling it.